src/components/TradeWidget/index.tsx

```diff
@@ -275,25 +217,45 @@ const TradeWidget: React.FC = () => {
     () =>
       getToken('symbol', receiveTokenSymbol, tokens) || (getToken('symbol', 'USDC', tokens) as Required<TokenDetails>),
   )
-  const [unlimited, setUnlimited] = useState(!validUntil || !Number(validUntil))
-  const sellInputId = TradeFormTokenId.sellToken
-  const receiveInputId = TradeFormTokenId.receiveToken
-  const validUntilId = TradeFormTokenId.validUntil
+  const [unlimited, setUnlimited] = useState(!validUntilParam || !Number(validUntilParam))
 
   const methods = useForm<TradeFormData>({
     mode: 'onChange',
     defaultValues: {
-      [sellInputId]: sellAmount,
-      [receiveInputId]: receiveAmount,
-      [validUntilId]: validUntil,
+      [sellInputId]: sellParam,
+      [receiveInputId]: '',
+      [validUntilId]: validUntilParam,
+      [priceInputId]: priceParam,
+      [priceInverseInputId]: '',
     },
   })
-  const { handleSubmit, reset, watch } = methods
+  const { handleSubmit, reset, watch, setValue } = methods
+
+  const priceValue = watch(priceInputId)
+  const priceInverseValue = watch(priceInverseInputId)
+  const sellValue = watch(sellInputId)
+
+  const receiveValue = watch(receiveInputId)
+  const validUntilValue = watch(validUntilId)
+
+  // Update receive amount
+  useEffect(() => {
+    let receiveAmount: string | undefined
+    if (priceValue && sellValue) {
+      const sellAmount = parseBigNumber(sellValue)
+      const price = parseBigNumber(priceValue)
+
+      if (sellAmount && price) {
+        receiveAmount = sellAmount.multipliedBy(price).toString()
```

Comment:
```suggestion
        receiveAmount = sellAmount.multipliedBy(price).toString(10)
```

Default BigNumber toString() uses exponential notation. Setting the base (10) forces it not to.

Unless we do want exponential notation in the price?

Examples:

![Screenshot from 2020-02-20 09-12-24](https://user-images.githubusercontent.com/43217/74932733-49ca2700-53c1-11ea-9895-8955c4267230.png)
![Screenshot from 2020-02-20 09-11-19](https://user-images.githubusercontent.com/43217/74932737-4b93ea80-53c1-11ea-92fc-d85eb95c8d66.png)

Reply:
Nop, decimal is better I think. Thanks for the tip!